### PR TITLE
Registers package with sunpy template

### PR DESCRIPTION
**Pull request by CyclingNinja**

### CODE REVIEW

File: .cruft.json [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+  "context": {
+    "cookiecutter": {
+      "package_name": "sunkit-instruments",
+      "module_name": "sunkit_instruments",
+      "short_description": "A SunPy-affiliated package for solar instrum",
+      "author_name": "The SunPy Community",
+      "author_email": "sunpy@googlegroups.com",
+      "project_url": "https://sunpy.org",
+      "license": "BSD 3-Clause",
+      "minimum_python_version": "3.9",
```
Comment (nabobalis) [inline .cruft.json:14]:
```suggestion
      "minimum_python_version": "3.10",
```

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -29,25 +29,41 @@ jobs:
...
+    secrets:
+      CODECOV_TOKEN: ${{ secrets.CODECOV_TOKEN }}
 
   docs:
-    needs: [test]
+    needs: [core]
     uses: OpenAstronomy/github-actions-workflows/.github/workflows/tox.yml@main
     with:
-      default_python: '3.12'
+      default_python: '3.10'
```
Comment (nabobalis) [inline .github/workflows/ci.yml:66]:
```suggestion
      default_python: '3.12'
```

File: docs/conf.py [hunk 3]
```diff
@@ -87,6 +106,9 @@
 
 # -- Options for HTML output ---------------------------------------------------
 html_theme = "sunpy"
+from sunpy_sphinx_theme import *
```
Comment (nabobalis) [inline docs/conf.py:109]:
```suggestion
from sunpy_sphinx_theme import PNG_ICON
```

File: docs/conf.py [hunk 4]
```diff
@@ -106,6 +128,7 @@
     "examples_dirs": os.path.join("..", "examples"),
     "gallery_dirs": os.path.join("generated", "gallery"),
     "matplotlib_animations": True,
+    # Comes from the theme.
```
Comment (nabobalis) [inline docs/conf.py:131]:
```suggestion
```

**Review by nabobalis — APPROVED** [review]

**Comment by Cadair** [comment]
Seriously what has upset minimums dependencies this much

**Comment by nabobalis** [comment]
> Seriously what has upset minimums dependencies this much

Setup.cfg being there, and the warnings being missing from pytest.

**Comment by nabobalis** [comment]
> Seriously what has upset minimums dependencies this much
```
ERROR: Ignored the following yanked versions: 1.11.0, 1.14.0rc1
ERROR: Ignored the following versions that require a different python version: 1.10.0 Requires-Python <3.12,>=3.8; 1.10.0rc1 Requires-Python <3.12,>=3.8; 1.10.0rc2 Requires-Python <3.12,>=3.8; 1.10.1 Requires-Python <3.12,>=3.8; 1.6.2 Requires-Python >=3.7,<3.10; 1.6.3 Requires-Python >=3.7,<3.10; 1.7.0 Requires-Python >=3.7,<3.10; 1.7.1 Requires-Python >=3.7,<3.10; 1.7.2 Requires-Python >=3.7,<3.11; 1.7.3 Requires-Python >=3.7,<3.11; 1.8.0 Requires-Python >=3.8,<3.11; 1.8.0rc1 Requires-Python >=3.8,<3.11; 1.8.0rc2 Requires-Python >=3.8,<3.11; 1.8.0rc3 Requires-Python >=3.8,<3.11; 1.8.0rc4 Requires-Python >=3.8,<3.11; 1.8.1 Requires-Python >=3.8,<3.11; 1.9.0 Requires-Python >=3.8,<3.12; 1.9.0rc1 Requires-Python >=3.8,<3.12; 1.9.0rc2 Requires-Python >=3.8,<3.12; 1.9.0rc3 Requires-Python >=3.8,<3.12; 1.9.1 Requires-Python >=3.8,<3.12
ERROR: Could not find a version that satisfies the requirement scipy==1.10.1 (from versions: 0.8.0, 0.9.0, 0.10.0, 0.10.1, 0.11.0, 0.12.0, 0.12.1, 0.13.0, 0.13.1, 0.13.2, 0.13.3, 0.14.0, 0.14.1, 0.15.0, 0.15.1, 0.16.0, 0.16.1, 0.17.0, 0.17.1, 0.18.0, 0.18.1, 0.19.0, 0.19.1, 1.0.0, 1.0.1, 1.1.0, 1.2.0, 1.2.1, 1.2.2, 1.2.3, 1.3.0, 1.3.1, 1.3.2, 1.3.3, 1.4.0, 1.4.1, 1.5.0, 1.5.1, 1.5.2, 1.5.3, 1.5.4, 1.6.0, 1.6.1, 1.9.2, 1.9.3, 1.11.0rc1, 1.11.0rc2, 1.11.1, 1.11.2, 1.11.3, 1.11.4, 1.12.0rc1, 1.12.0rc2, 1.12.0, 1.13.0rc1, 1.13.0, 1.13.1, 1.14.0rc2, 1.14.0, 1.14.1)
ERROR: No matching distribution found for scipy==1.10.1
oldestdeps: exit 1 (0.95 seconds) /home/nabil/Git/sunkit-instruments/.tmp/oldestdeps> pip install -r requirements-min.txt pid=161460
  oldestdeps: FAIL code 1 (39.82=setup[36.16]+cmd[2.71,0.95] seconds)
  evaluation failed :( (39.88 seconds)
```

I thought it was due to setup.cfg being there but turns out that didnt fix it.

**Pull request merged**